students/navdeep/session06/mailroom_part4NoTest.py

```diff
@@ -0,0 +1,272 @@
+    values. The donation amounts are stored in lists. A donor can have more than
+    1 donation. Each donor is stored only once in the dictionary
+    :return: the initialized dictionary of donors and their donation amounts
+    """
+    donor_db = {"Navdeep Gill": [1000], "Nick Garzon": [5,2], "Henry Chipman": [100],
+                "Lorenzo B": [1000], "Torin Stetina": [200.75]}
+    return donor_db
+
+def initializeOptionsDB():
+    options_dict = {'t': askUserForDonor, 's': writeToFile, 'r': createReport, 'e': exitMessage}
+    return options_dict
+
+def displayMenu():
+    """
+    displays the menu of options to the user
+    """
+    print()
+    str_menu = "1. Type 't' to Send a Thank You to add a name, donation and thank you card\n"  \
+               "2. Type 'r' to Create a Report and see a list of the donors\n" \
+               "3. Type 's' to Send letters to everyone\n" \
+               "4. Type 'e' to exit the program"
+
+    #for testing purposes to determine menu is returned correctly
+    #return str_menu
+    print(str_menu)
+    print()
+
+def menu(donor_dict, menu_dict):
+    """This function will print out a menu of options for
+    the user to choose from
+    :param: the list of donors and how much they have donated"""
+    user_option = ""
+    while user_option.lower() != 'e':
+        displayMenu()
+        #user_option = input("Select an option from the menu: ")
+        user_option = 't' #for testing purposes
+        user_option = user_option.strip()
+        user_option = user_option.lower()
+        return user_option #used to test menu produces correct user input
+        #try:
+         #   menu_dict[user_option](donor_dict)
+        #except KeyError as e:
+         #   print("Invalid option selected. Please try again")
+
+def askUserForDonor(donor_list):
+    """
+    Allows user to add a new donor to the dictionary or print out
+    the names of the current donors or exit back to the main menu.
+    The user can also type in the name of a donor that already
+    exists to append a new donation.
+    :param donor_list: the list of donors and their donations
+    """
+    user_donor_option = ""
+    while user_donor_option != 'exit':
+        user_donor_option = input("1. Type 'new' of a new donor\n"
+                                  "2. Type 'list' to list the donor names\n"
+                                  "3. Type 'exit' to return to main menu\n"
+                                  "Your Selection:  ")
+        print()
+
+        user_donor_option = user_donor_option.strip().lower()
+        if user_donor_option == 'new':
+            addNewDonor(donor_list)
+        elif user_donor_option == 'list':
+            showDonorNames(donor_list)
+        elif user_donor_option == 'exit':
+            return
+        else:
+            print("Please select a valid option:")
+            print()
+
+def addNewDonor(donor_dict):
+    first_name = None
```

Comment:
You can force these to lower and use strip here so that you only need to do that in one place.

Reply:
Hey Summer,

Thank for the comment!  Do you mean in regards to the "first_name" variable I can use the lower and strip functions when I declare the variable?